util/chplenv/utils.py

```diff
@@ -107,3 +108,78 @@ def run_live_command(command):
 
     if returncode != 0:
         error("command failed: {0}".format(command), CommandError)
+
+
+def _split_ver_str(ver_str):
+    """Split a version string into an array of integers"""
+
+    # Semantic versioning strings can have prelease information
+    # after a hypen and/or build metadata after a plus. For
+    # the purpose of our versioning checks we strip these
+    ver_str = ver_str.rsplit('-',1)[0]
+    ver_str = ver_str.rsplit('+',1)[0]
+
+    return [int(x) for x in ver_str.split('.')]
+
+
+def is_ver_in_range(versionStr, minimumStr, maximumStr):
+    """Assume that version, minimum, and maximum are version strings consisting
+    of integers separated by spaces. Ensure that version is withint the
+    (inclusive) bounds: [minimumStr, maximumStr]."""
```

Comment:
Should also say within the bounds: [minimumStr, maximumStr) (non inclusive at the max).